### PR TITLE
Set content length for S3 client #1471

### PR DESCRIPTION
**Pull request by Jeeppler**
Set content length for S3 client #1471 and introduce `x-file-size` header #1551

- implemented set content length for job storage and s3 job storage #1471
- introduce `x-file-size` parameter #1551
- added/extended unit tests
- use the transfer manager #1573 

closes: #1471
closes: #1551
closes: #1573

### CODE REVIEW

File: sechub-testframework/src/main/java/com/mercedesbenz/sechub/test/TestUtil.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public static Path createTempDirectoryInBuildFolder(String dirName, FileAttribut
      *
      * @param explicitFileName the EXACT file name to use.
      *
-     * @return file
+     * @return filePath
```
Comment (de-jcup) [inline sechub-testframework/src/main/java/com/mercedesbenz/sechub/test/TestUtil.java:168]: Please change back to "file" - Path is the java.nio representation of an file or folder..

**Review by de-jcup — CHANGES_REQUESTED** [review]
Good job, but I want some changes  - please read the comments

**Review by winzj — CHANGES_REQUESTED** [review]
Good job, looks good to me. I have some minor suggestions, please have a look at my comments.

File: sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/internal/TestRestHelper.java [hunk 2]
```diff
@@ -264,6 +265,10 @@ public String upload(String uploadUrl, File file, String checkSum) {
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.MULTIPART_FORM_DATA);
 
+        headers.set("", checkSum);
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/internal/TestRestHelper.java:268]: I do not know, that was not my design decision.

@de-jcup can probably answer this question.

File: sechub-storage-s3-aws/src/main/java/com/mercedesbenz/sechub/storage/s3/AwsS3TransferManagerFactory.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.mercedesbenz.sechub.storage.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+
+public class AwsS3TransferManagerFactory {
+    public TransferManager create(AmazonS3 client) {
+        long multipartUploadThreshold = 5 * 1024 * 1025; // threshold 5 MiB
```
Comment (winzj) [inline sechub-storage-s3-aws/src/main/java/com/mercedesbenz/sechub/storage/s3/AwsS3TransferManagerFactory.java:9]: I think there is a typo in this line.
```suggestion
        long multipartUploadThreshold = 5 * 1024 * 1024; // threshold 5 MiB
```

**Review by winzj — CHANGES_REQUESTED** [review]
Looks good, just found a small typo in a byte size definition. Please, see comment below.

**Review by winzj — APPROVED** [review]
Looks good.

**Pull request merged**